src/shared/modules/logger/logger-factory.ts

```diff
@@ -0,0 +1,108 @@
+import { ConfigService } from '@nestjs/config';
+import { IncomingMessage, ServerResponse } from 'http';
+import { Params } from 'nestjs-pino';
+import { GenReqId, Options } from 'pino-http';
+import { ulid } from 'ulid';
+
+const genReqId: GenReqId = (
+  req: IncomingMessage,
+  res: ServerResponse<IncomingMessage>,
+) => {
+  const headerId = req.headers['x-request-id'];
+  const id = typeof headerId === 'string' && headerId ? headerId : ulid();
+  res.setHeader('X-Request-Id', String(id));
+  return id;
+};
+
+const customReceivedMessage = (req: IncomingMessage): string => {
+  const reqId = String((req as any).id || '*');
+  return `[${reqId}] => Request received: "${req.method} ${req.url}"`;
+};
+
+const customSuccessMessage = (
+  req: IncomingMessage,
+  res: ServerResponse<IncomingMessage>,
+  responseTime: number,
+): string => {
+  const reqId = String((req as any).id || '*');
+  return `[${reqId}] "${req.method} ${req.url}" ${res.statusCode} - ${responseTime} ms`;
+};
+
+const customErrorMessage = (
+  req: IncomingMessage,
+  res: ServerResponse<IncomingMessage>,
+  err: Error,
+): string => {
+  const reqId = String((req as any).id || '*');
+  return `[${reqId}] "${req.method} ${req.url}" ${res.statusCode} - Error: ${err.message}`;
+};
```

Comment:
Could you avoid this any here ? nestjs and node have correct types to request and responses.